pkg/controller/common/daemonset/reconcile.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package daemonset
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/hash"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+)
+
+func New(podTemplate corev1.PodTemplateSpec, name string, labels map[string]string, owner metav1.Object, selectors map[string]string) appsv1.DaemonSet {
+	return appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: owner.GetNamespace(),
+			Labels:    labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: selectors,
+			},
+			Template: podTemplate,
+		},
+	}
+}
+
+// Reconcile creates or updates the given daemon set for the specified owner.
+func Reconcile(
+	k8sClient k8s.Client,
+	expected appsv1.DaemonSet,
+	owner metav1.Object,
+) (appsv1.DaemonSet, error) {
+	// label the daemon set with a hash of itself
+	expected = WithTemplateHash(expected)
+
+	reconciled := &appsv1.DaemonSet{}
+	err := reconciler.ReconcileResource(reconciler.Params{
+		Client:     k8sClient,
+		Owner:      owner,
+		Expected:   &expected,
+		Reconciled: reconciled,
+		NeedsUpdate: func() bool {
+			// compare hash of the daemon set at the time it was built
```

Comment:
```suggestion
			// compare hash of the DaemonSet at the time it was built
```
also not sure I fully understand the comment